# Importation de la bobliothèque

In [3]:
import pandas as pd

In [ ]:
# importation du dataset

In [6]:
df = pd.read_csv("student-mat.csv" , sep = ";")
df

school sex  age address famsize Pstatus  Medu  Fedu      Mjob      Fjob  \
0       GP   F   18       U     GT3       A     4     4   at_home   teacher   
1       GP   F   17       U     GT3       T     1     1   at_home     other   
2       GP   F   15       U     LE3       T     1     1   at_home     other   
3       GP   F   15       U     GT3       T     4     2    health  services   
4       GP   F   16       U     GT3       T     3     3     other     other   
..     ...  ..  ...     ...     ...     ...   ...   ...       ...       ...   
390     MS   M   20       U     LE3       A     2     2  services  services   
391     MS   M   17       U     LE3       T     3     1  services  services   
392     MS   M   21       R     GT3       T     1     1     other     other   
393     MS   M   18       R     LE3       T     3     2  services     other   
394     MS   M   19       U     LE3       T     1     1     other   at_home   

     ... famrel freetime  goout  Dalc  Walc health absences  G1  G2  G3  
0    ...      4        3      4     1     1      3        6   5   6   6  
1    ...      5        3      3     1     1      3        4   5   5   6  
2    ...      4        3      2     2     3      3       10   7   8  10  
3    ...      3        2      2     1     1      5        2  15  14  15  
4    ...      4        3      2     1     2      5        4   6  10  10  
..   ...    ...      ...    ...   ...   ...    ...      ...  ..  ..  ..  
390  ...      5        5      4     4     5      4       11   9   9   9  
391  ...      2        4      5     3     4      2        3  14  16  16  
392  ...      5        5      3     3     3      3        3  10   8   7  
393  ...      4        4      1     3     4      5        0  11  12  10  
394  ...      3        2      3     3     3      5        5   8   9   9  

[395 rows x 33 columns]

# origine: collecte des données à caractère personnels sur les etudiants
# circonstances de création : descente au niveau des établissements pour recueillir les informations de facon virtuelle ou reeles
# prévisions sur le dataset : prévisions des notes, du rang social des etudiants
# bref apercu : les données sont representés sous formes matricielle  de dimmension 395*33 et a premieres vues ne contient pas de valeurs manquantes.

# question 3: les 10premieres lignes du dataset

In [8]:
df.head(10)

school sex  age address famsize Pstatus  Medu  Fedu      Mjob      Fjob  \
0     GP   F   18       U     GT3       A     4     4   at_home   teacher   
1     GP   F   17       U     GT3       T     1     1   at_home     other   
2     GP   F   15       U     LE3       T     1     1   at_home     other   
3     GP   F   15       U     GT3       T     4     2    health  services   
4     GP   F   16       U     GT3       T     3     3     other     other   
5     GP   M   16       U     LE3       T     4     3  services     other   
6     GP   M   16       U     LE3       T     2     2     other     other   
7     GP   F   17       U     GT3       A     4     4     other   teacher   
8     GP   M   15       U     LE3       A     3     2  services     other   
9     GP   M   15       U     GT3       T     3     4     other     other   

   ... famrel freetime  goout  Dalc  Walc health absences  G1  G2  G3  
0  ...      4        3      4     1     1      3        6   5   6   6  
1  ...      5        3      3     1     1      3        4   5   5   6  
2  ...      4        3      2     2     3      3       10   7   8  10  
3  ...      3        2      2     1     1      5        2  15  14  15  
4  ...      4        3      2     1     2      5        4   6  10  10  
5  ...      5        4      2     1     2      5       10  15  15  15  
6  ...      4        4      4     1     1      3        0  12  12  11  
7  ...      4        1      4     1     1      1        6   6   5   6  
8  ...      4        2      2     1     1      1        0  16  18  19  
9  ...      5        5      1     1     1      5        0  14  15  15  

[10 rows x 33 columns]

# question 4: les 10premieres colonnes du dataset

In [9]:
df.tail(10)

school sex  age address famsize Pstatus  Medu  Fedu      Mjob      Fjob  \
385     MS   F   18       R     GT3       T     2     2   at_home     other   
386     MS   F   18       R     GT3       T     4     4   teacher   at_home   
387     MS   F   19       R     GT3       T     2     3  services     other   
388     MS   F   18       U     LE3       T     3     1   teacher  services   
389     MS   F   18       U     GT3       T     1     1     other     other   
390     MS   M   20       U     LE3       A     2     2  services  services   
391     MS   M   17       U     LE3       T     3     1  services  services   
392     MS   M   21       R     GT3       T     1     1     other     other   
393     MS   M   18       R     LE3       T     3     2  services     other   
394     MS   M   19       U     LE3       T     1     1     other   at_home   

     ... famrel freetime  goout  Dalc  Walc health absences  G1  G2  G3  
385  ...      5        3      3     1     3      4        2  10   9  10  
386  ...      4        4      3     2     2      5        7   6   5   6  
387  ...      5        4      2     1     2      5        0   7   5   0  
388  ...      4        3      4     1     1      1        0   7   9   8  
389  ...      1        1      1     1     1      5        0   6   5   0  
390  ...      5        5      4     4     5      4       11   9   9   9  
391  ...      2        4      5     3     4      2        3  14  16  16  
392  ...      5        5      3     3     3      3        3  10   8   7  
393  ...      4        4      1     3     4      5        0  11  12  10  
394  ...      3        2      3     3     3      5        5   8   9   9  

[10 rows x 33 columns]

# question 5: les colonnes du dataset

In [10]:
df.columns

Index(['school', 'sex', 'age', 'address', 'famsize', 'Pstatus', 'Medu', 'Fedu',
       'Mjob', 'Fjob', 'reason', 'guardian', 'traveltime', 'studytime',
       'failures', 'schoolsup', 'famsup', 'paid', 'activities', 'nursery',
       'higher', 'internet', 'romantic', 'famrel', 'freetime', 'goout', 'Dalc',
       'Walc', 'health', 'absences', 'G1', 'G2', 'G3'],
      dtype='object')

# question 6: colonne du dataset en majuscules

In [11]:
df.columns.str.upper()

Index(['SCHOOL', 'SEX', 'AGE', 'ADDRESS', 'FAMSIZE', 'PSTATUS', 'MEDU', 'FEDU',
       'MJOB', 'FJOB', 'REASON', 'GUARDIAN', 'TRAVELTIME', 'STUDYTIME',
       'FAILURES', 'SCHOOLSUP', 'FAMSUP', 'PAID', 'ACTIVITIES', 'NURSERY',
       'HIGHER', 'INTERNET', 'ROMANTIC', 'FAMREL', 'FREETIME', 'GOOUT', 'DALC',
       'WALC', 'HEALTH', 'ABSENCES', 'G1', 'G2', 'G3'],
      dtype='object')

# question 7: taille du dataset

In [12]:
df.shape

(395, 33)

# question 8: informations sur le type de colonnes

In [16]:
df.info() # avec ca on a les informations sur les types de colonnes

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 395 entries, 0 to 394
Data columns (total 33 columns):
 #   Column      Non-Null Count  Dtype 
---  ------      --------------  ----- 
 0   school      395 non-null    object
 1   sex         395 non-null    object
 2   age         395 non-null    int64 
 3   address     395 non-null    object
 4   famsize     395 non-null    object
 5   Pstatus     395 non-null    object
 6   Medu        395 non-null    int64 
 7   Fedu        395 non-null    int64 
 8   Mjob        395 non-null    object
 9   Fjob        395 non-null    object
 10  reason      395 non-null    object
 11  guardian    395 non-null    object
 12  traveltime  395 non-null    int64 
 13  studytime   395 non-null    int64 
 14  failures    395 non-null    int64 
 15  schoolsup   395 non-null    object
 16  famsup      395 non-null    object
 17  paid        395 non-null    object
 18  activities  395 non-null    object
 19  nursery     395 non-null    object
 20  higher    

# on peut déduire d'ici qu'il y a pas de valeurs manquantes

# types de données

In [17]:
df.dtypes # avec ca on a l'information sur les types de colonnes

school        object
sex           object
age            int64
address       object
famsize       object
Pstatus       object
Medu           int64
Fedu           int64
Mjob          object
Fjob          object
reason        object
guardian      object
traveltime     int64
studytime      int64
failures       int64
schoolsup     object
famsup        object
paid          object
activities    object
nursery       object
higher        object
internet      object
romantic      object
famrel         int64
freetime       int64
goout          int64
Dalc           int64
Walc           int64
health         int64
absences       int64
G1             int64
G2             int64
G3             int64
dtype: object

In [18]:
df.isnull()

school    sex    age  address  famsize  Pstatus   Medu   Fedu   Mjob  \
0     False  False  False    False    False    False  False  False  False   
1     False  False  False    False    False    False  False  False  False   
2     False  False  False    False    False    False  False  False  False   
3     False  False  False    False    False    False  False  False  False   
4     False  False  False    False    False    False  False  False  False   
..      ...    ...    ...      ...      ...      ...    ...    ...    ...   
390   False  False  False    False    False    False  False  False  False   
391   False  False  False    False    False    False  False  False  False   
392   False  False  False    False    False    False  False  False  False   
393   False  False  False    False    False    False  False  False  False   
394   False  False  False    False    False    False  False  False  False   

      Fjob  ...  famrel  freetime  goout   Dalc   Walc  health  absences  \
0    False  ...   False     False  False  False  False   False     False   
1    False  ...   False     False  False  False  False   False     False   
2    False  ...   False     False  False  False  False   False     False   
3    False  ...   False     False  False  False  False   False     False   
4    False  ...   False     False  False  False  False   False     False   
..     ...  ...     ...       ...    ...    ...    ...     ...       ...   
390  False  ...   False     False  False  False  False   False     False   
391  False  ...   False     False  False  False  False   False     False   
392  False  ...   False     False  False  False  False   False     False   
393  False  ...   False     False  False  False  False   False     False   
394  False  ...   False     False  False  False  False   False     False   

        G1     G2     G3  
0    False  False  False  
1    False  False  False  
2    False  False  False  
3    False  False  False  
4    False  False  False  
..     ...    ...    ...  
390  False  False  False  
391  False  False  False  
392  False  False  False  
393  False  False  False  
394  False  False  False  

[395 rows x 33 columns]

In [19]:
df.isnull().sum() # pour avoir les valeurs manquantes par colonnes

school        0
sex           0
age           0
address       0
famsize       0
Pstatus       0
Medu          0
Fedu          0
Mjob          0
Fjob          0
reason        0
guardian      0
traveltime    0
studytime     0
failures      0
schoolsup     0
famsup        0
paid          0
activities    0
nursery       0
higher        0
internet      0
romantic      0
famrel        0
freetime      0
goout         0
Dalc          0
Walc          0
health        0
absences      0
G1            0
G2            0
G3            0
dtype: int64

# il n'y a pas de valeur manquantes

# question 9: statistique sur les colonnes

In [22]:
df.describe()

age        Medu        Fedu  traveltime   studytime    failures  \
count  395.000000  395.000000  395.000000  395.000000  395.000000  395.000000   
mean    16.696203    2.749367    2.521519    1.448101    2.035443    0.334177   
std      1.276043    1.094735    1.088201    0.697505    0.839240    0.743651   
min     15.000000    0.000000    0.000000    1.000000    1.000000    0.000000   
25%     16.000000    2.000000    2.000000    1.000000    1.000000    0.000000   
50%     17.000000    3.000000    2.000000    1.000000    2.000000    0.000000   
75%     18.000000    4.000000    3.000000    2.000000    2.000000    0.000000   
max     22.000000    4.000000    4.000000    4.000000    4.000000    3.000000   

           famrel    freetime       goout        Dalc        Walc      health  \
count  395.000000  395.000000  395.000000  395.000000  395.000000  395.000000   
mean     3.944304    3.235443    3.108861    1.481013    2.291139    3.554430   
std      0.896659    0.998862    1.113278    0.890741    1.287897    1.390303   
min      1.000000    1.000000    1.000000    1.000000    1.000000    1.000000   
25%      4.000000    3.000000    2.000000    1.000000    1.000000    3.000000   
50%      4.000000    3.000000    3.000000    1.000000    2.000000    4.000000   
75%      5.000000    4.000000    4.000000    2.000000    3.000000    5.000000   
max      5.000000    5.000000    5.000000    5.000000    5.000000    5.000000   

         absences          G1          G2          G3  
count  395.000000  395.000000  395.000000  395.000000  
mean     5.708861   10.908861   10.713924   10.415190  
std      8.003096    3.319195    3.761505    4.581443  
min      0.000000    3.000000    0.000000    0.000000  
25%      0.000000    8.000000    9.000000    8.000000  
50%      4.000000   11.000000   11.000000   11.000000  
75%      8.000000   13.000000   13.000000   14.000000  
max     75.000000   19.000000   19.000000   20.000000

In [ ]:
# interpretation , cette partie contient trois sous parties

# nombre et types de colonnes indexées : 16 colonnes codées sur 64 bits et de types entiers

# corelation obtenu : forte corelation entre les données de chaque colonnes. les donnees sont concentrées

#interpretation sur la relation entre la valeurs maximales et la moyenne : (max/count)/16. et comme le resultat est tres proche de la moyenne, on peut dire que le données fournies ont ete fiables

# question 10 : transposition du dataset

In [24]:
df.transpose()

0        1        2         3       4           5       6    \
school           GP       GP       GP        GP      GP          GP      GP   
sex               F        F        F         F       F           M       M   
age              18       17       15        15      16          16      16   
address           U        U        U         U       U           U       U   
famsize         GT3      GT3      LE3       GT3     GT3         LE3     LE3   
Pstatus           A        T        T         T       T           T       T   
Medu              4        1        1         4       3           4       2   
Fedu              4        1        1         2       3           3       2   
Mjob        at_home  at_home  at_home    health   other    services   other   
Fjob        teacher    other    other  services   other       other   other   
reason       course   course    other      home    home  reputation    home   
guardian     mother   father   mother    mother  father      mother  mother   
traveltime        2        1        1         1       1           1       1   
studytime         2        2        2         3       2           2       2   
failures          0        0        3         0       0           0       0   
schoolsup       yes       no      yes        no      no          no      no   
famsup           no      yes       no       yes     yes         yes      no   
paid             no       no      yes       yes     yes         yes      no   
activities       no       no       no       yes      no         yes      no   
nursery         yes       no      yes       yes     yes         yes     yes   
higher          yes      yes      yes       yes     yes         yes     yes   
internet         no      yes      yes       yes      no         yes     yes   
romantic         no       no       no       yes      no          no      no   
famrel            4        5        4         3       4           5       4   
freetime          3        3        3         2       3           4       4   
goout             4        3        2         2       2           2       4   
Dalc              1        1        2         1       1           1       1   
Walc              1        1        3         1       2           2       1   
health            3        3        3         5       5           5       3   
absences          6        4       10         2       4          10       0   
G1                5        5        7        15       6          15      12   
G2                6        5        8        14      10          15      12   
G3                6        6       10        15      10          15      11   

                7         8       9    ...      385         386       387  \
school           GP        GP      GP  ...       MS          MS        MS   
sex               F         M       M  ...        F           F         F   
age              17        15      15  ...       18          18        19   
address           U         U       U  ...        R           R         R   
famsize         GT3       LE3     GT3  ...      GT3         GT3       GT3   
Pstatus           A         A       T  ...        T           T         T   
Medu              4         3       3  ...        2           4         2   
Fedu              4         2       4  ...        2           4         3   
Mjob          other  services   other  ...  at_home     teacher  services   
Fjob        teacher     other   other  ...    other     at_home     other   
reason         home      home    home  ...    other  reputation    course   
guardian     mother    mother  mother  ...   mother      mother    mother   
traveltime        2         1       1  ...        2           3         1   
studytime         2         2       2  ...        3           1         3   
failures          0         0       0  ...        0           0         1   
schoolsup       yes        no      no  ...       no          no        no   
famsup          yes 

In [25]:
df.T

0        1        2         3       4           5       6    \
school           GP       GP       GP        GP      GP          GP      GP   
sex               F        F        F         F       F           M       M   
age              18       17       15        15      16          16      16   
address           U        U        U         U       U           U       U   
famsize         GT3      GT3      LE3       GT3     GT3         LE3     LE3   
Pstatus           A        T        T         T       T           T       T   
Medu              4        1        1         4       3           4       2   
Fedu              4        1        1         2       3           3       2   
Mjob        at_home  at_home  at_home    health   other    services   other   
Fjob        teacher    other    other  services   other       other   other   
reason       course   course    other      home    home  reputation    home   
guardian     mother   father   mother    mother  father      mother  mother   
traveltime        2        1        1         1       1           1       1   
studytime         2        2        2         3       2           2       2   
failures          0        0        3         0       0           0       0   
schoolsup       yes       no      yes        no      no          no      no   
famsup           no      yes       no       yes     yes         yes      no   
paid             no       no      yes       yes     yes         yes      no   
activities       no       no       no       yes      no         yes      no   
nursery         yes       no      yes       yes     yes         yes     yes   
higher          yes      yes      yes       yes     yes         yes     yes   
internet         no      yes      yes       yes      no         yes     yes   
romantic         no       no       no       yes      no          no      no   
famrel            4        5        4         3       4           5       4   
freetime          3        3        3         2       3           4       4   
goout             4        3        2         2       2           2       4   
Dalc              1        1        2         1       1           1       1   
Walc              1        1        3         1       2           2       1   
health            3        3        3         5       5           5       3   
absences          6        4       10         2       4          10       0   
G1                5        5        7        15       6          15      12   
G2                6        5        8        14      10          15      12   
G3                6        6       10        15      10          15      11   

                7         8       9    ...      385         386       387  \
school           GP        GP      GP  ...       MS          MS        MS   
sex               F         M       M  ...        F           F         F   
age              17        15      15  ...       18          18        19   
address           U         U       U  ...        R           R         R   
famsize         GT3       LE3     GT3  ...      GT3         GT3       GT3   
Pstatus           A         A       T  ...        T           T         T   
Medu              4         3       3  ...        2           4         2   
Fedu              4         2       4  ...        2           4         3   
Mjob          other  services   other  ...  at_home     teacher  services   
Fjob        teacher     other   other  ...    other     at_home     other   
reason         home      home    home  ...    other  reputation    course   
guardian     mother    mother  mother  ...   mother      mother    mother   
traveltime        2         1       1  ...        2           3         1   
studytime         2         2       2  ...        3           1         3   
failures          0         0       0  ...        0           0         1   
schoolsup       yes        no      no  ...       no          no        no   
famsup          yes 

# question 11 : colonne age du dataset

In [26]:
A = df['age']
A

0      18
1      17
2      15
3      15
4      16
       ..
390    20
391    17
392    21
393    18
394    19
Name: age, Length: 395, dtype: int64

In [ ]:
# question 12 : Moyenne colonne age du dataset

In [27]:
A.mean()

16.696202531645568

In [28]:
A.max()

22

In [29]:
A.min()

15

In [30]:
A.std() #ecart type

1.2760427246056245

In [32]:
A.describe()

count    395.000000
mean      16.696203
std        1.276043
min       15.000000
25%       16.000000
50%       17.000000
75%       18.000000
max       22.000000
Name: age, dtype: float64

# 14- Extraction

In [33]:
A = df['sex']
B = df ['age']
C = df ['studytime']
D = df ['G1']
E = df ['G2']
F = df ['G3']

Bd1 = pd.concat([A, B, C, D, E, F], axis = 1)
Bd1

sex  age  studytime  G1  G2  G3
0     F   18          2   5   6   6
1     F   17          2   5   5   6
2     F   15          2   7   8  10
3     F   15          3  15  14  15
4     F   16          2   6  10  10
..   ..  ...        ...  ..  ..  ..
390   M   20          2   9   9   9
391   M   17          1  14  16  16
392   M   21          1  10   8   7
393   M   18          1  11  12  10
394   M   19          1   8   9   9

[395 rows x 6 columns]

# 15- Affichons les stats de cette mini bd

In [34]:
Bd1.describe() 

age   studytime          G1          G2          G3
count  395.000000  395.000000  395.000000  395.000000  395.000000
mean    16.696203    2.035443   10.908861   10.713924   10.415190
std      1.276043    0.839240    3.319195    3.761505    4.581443
min     15.000000    1.000000    3.000000    0.000000    0.000000
25%     16.000000    1.000000    8.000000    9.000000    8.000000
50%     17.000000    2.000000   11.000000   11.000000   11.000000
75%     18.000000    2.000000   13.000000   13.000000   14.000000
max     22.000000    4.000000   19.000000   19.000000   20.000000

# ca n'affiche pas les sexe parceque la fonction describe() ne tient compte que des valeurs quantitatives

# 16- nombre de garcons et de filles de la bd

In [35]:
A = df ['sex']
A

0      F
1      F
2      F
3      F
4      F
      ..
390    M
391    M
392    M
393    M
394    M
Name: sex, Length: 395, dtype: object

In [36]:
A.value_counts()

F    208
M    187
Name: sex, dtype: int64

In [38]:
femme = Bd1["sex"] == "F"
female = Bd1[femme]
female
len(Bd1[femme])

208

In [39]:
homme = Bd1["sex"] == "M"
Male = Bd1[homme]
Male
len(Bd1[homme])

187

# 17- Affichons les infos sur les etudiants a indice 4

In [42]:
df [4:5]

school sex  age address famsize Pstatus  Medu  Fedu   Mjob   Fjob  ...  \
4     GP   F   16       U     GT3       T     3     3  other  other  ...   

  famrel freetime  goout  Dalc  Walc health absences G1  G2  G3  
4      4        3      2     1     2      5        4  6  10  10  

[1 rows x 33 columns]

# lorsqu'on manipule les lignes le ilock est deconseiller

# 18- selection mini bd

In [44]:
a = df [10:20]
a

school sex  age address famsize Pstatus  Medu  Fedu      Mjob      Fjob  \
10     GP   F   15       U     GT3       T     4     4   teacher    health   
11     GP   F   15       U     GT3       T     2     1  services     other   
12     GP   M   15       U     LE3       T     4     4    health  services   
13     GP   M   15       U     GT3       T     4     3   teacher     other   
14     GP   M   15       U     GT3       A     2     2     other     other   
15     GP   F   16       U     GT3       T     4     4    health     other   
16     GP   F   16       U     GT3       T     4     4  services  services   
17     GP   F   16       U     GT3       T     3     3     other     other   
18     GP   M   17       U     GT3       T     3     2  services  services   
19     GP   M   16       U     LE3       T     4     3    health     other   

    ... famrel freetime  goout  Dalc  Walc health absences  G1  G2  G3  
10  ...      3        3      3     1     2      2        0  10   8   9  
11  ...      5        2      2     1     1      4        4  10  12  12  
12  ...      4        3      3     1     3      5        2  14  14  14  
13  ...      5        4      3     1     2      3        2  10  10  11  
14  ...      4        5      2     1     1      3        0  14  16  16  
15  ...      4        4      4     1     2      2        4  14  14  14  
16  ...      3        2      3     1     2      2        6  13  14  14  
17  ...      5        3      2     1     1      4        4   8  10  10  
18  ...      5        5      5     2     4      5       16   6   5   5  
19  ...      3        1      3     1     3      5        4   8  10  10  

[10 rows x 33 columns]

# 19- affichages des notes

In [46]:
D = a ['G1']
E = a ['G2']
F = a ['G3']
Bd2 = pd.concat([D,E,F], axis=1)
Bd2

G1  G2  G3
10  10   8   9
11  10  12  12
12  14  14  14
13  10  10  11
14  14  16  16
15  14  14  14
16  13  14  14
17   8  10  10
18   6   5   5
19   8  10  10

# 20- statistique sur cette selection

In [47]:
Bd2.describe()

G1      G2         G3
count  10.000000  10.000  10.000000
mean   10.700000  11.300  11.500000
std     2.907844   3.335   3.205897
min     6.000000   5.000   5.000000
25%     8.500000  10.000  10.000000
50%    10.000000  11.000  11.500000
75%    13.750000  14.000  14.000000
max    14.000000  16.000  16.000000

# 21- affichons les notes des etudiantes 

In [48]:
femme = Bd1['sex'] == "F"
female = Bd1[femme]
female
len(Bd1[femme])

208

In [50]:
D = female ['G1']
E = female ['G2']
F = female ['G3']
Bd3 = pd.concat([D,E,F], axis=1)
Bd3

G1  G2  G3
0     5   6   6
1     5   5   6
2     7   8  10
3    15  14  15
4     6  10  10
..   ..  ..  ..
385  10   9  10
386   6   5   6
387   7   5   0
388   7   9   8
389   6   5   0

[208 rows x 3 columns]

# 22- Affichons les notes et l'age des etudiantes qui etudies 2h par jour

In [58]:
df3=df[df["sex"]=="F"]
df5 = df3 [df3["studytime"]==2]
x = df5["G1"]
y = df5["G2"]
z = df5["G3"]
t = df5["age"]
df6 = pd.concat([x,y,z,t], axis = 1)
df6

G1  G2  G3  age
0     5   6   6   18
1     5   5   6   17
2     7   8  10   15
4     6  10  10   16
7     6   5   6   17
..   ..  ..  ..  ...
377   8   9  10   18
378  15  15  15   18
379  10  10  10   17
388   7   9   8   18
389   6   5   0   18

[113 rows x 4 columns]

# 23-Affichage des occurences des metiers des papas

In [60]:
df["Fjob"].value_counts()

other       217
services    111
teacher      29
at_home      20
health       18
Name: Fjob, dtype: int64

# 24-Affichage des notes des etudiantes dont le papa est medecin ou enseignant

In [62]:
health = df3 [df3["Fjob"] == "health"]
teacher = df3 [df3 ["Fjob"] == "teacher"]
k = teacher.append(health)
x = k["G1"]
y = k["G2"]
z = k["G3"]
t = k["age"]
df6 = pd.concat([x,y,z,t], axis = 1)
df6

school sex  age address famsize Pstatus  Medu  Fedu      Mjob     Fjob  \
0       GP   F   18       U     GT3       A     4     4   at_home  teacher   
7       GP   F   17       U     GT3       A     4     4     other  teacher   
49      GP   F   15       U     GT3       T     4     4  services  teacher   
60      GP   F   16       R     GT3       T     4     4    health  teacher   
130     GP   F   15       R     GT3       T     3     4  services  teacher   
139     GP   F   15       U     GT3       T     4     4   teacher  teacher   
154     GP   F   17       U     GT3       T     4     4     other  teacher   
198     GP   F   17       U     GT3       T     4     4  services  teacher   
199     GP   F   16       U     LE3       T     4     4   teacher  teacher   
222     GP   F   16       U     GT3       T     2     3  services  teacher   
287     GP   F   17       U     GT3       T     1     1  services  teacher   
329     GP   F   17       U     GT3       T     4     4   teacher  teacher   
369     MS   F   18       R     GT3       T     4     4     other  teacher   
10      GP   F   15       U     GT3       T     4     4   teacher   health   
24      GP   F   15       R     GT3       T     2     4  services   health   
38      GP   F   15       R     GT3       T     3     4  services   health   
63      GP   F   16       U     GT3       T     4     3   teacher   health   
105     GP   F   15       U     GT3       A     3     3     other   health   
109     GP   F   16       U     LE3       T     4     4    health   health   
122     GP   F   16       U     LE3       T     2     4     other   health   
169     GP   F   16       U     GT3       T     4     4    health   health   
274     GP   F   17       U     GT3       T     2     4   at_home   health   
278     GP   F   18       U     GT3       T     4     4    health   health   
303     GP   F   17       U     GT3       T     3     2    health   health   
314     GP   F   19       U     GT3       T     1     1   at_home   health   

     ... famrel freetime  goout  Dalc  Walc health absences  G1  G2  G3  
0    ...      4        3      4     1     1      3        6   5   6   6  
7    ...      4        1      4     1     1      1        6   6   5   6  
49   ...      4        4      4     1     1      3        2   7   7   7  
60   ...      2        4      4     2     3      4        6  10  11  11  
130  ...      4        2      2     2     2      5        0  12   0   0  
139  ...      4        3      2     1     1      5        0  16  16  15  
154  ...      4        2      1     1     1      4        0  11  11  12  
198  ...      4        2      4     2     3      2       24  18  18  18  
199  ...      4        5      2     1     2      3        0   9   9  10  
222  ...      2        3      1     1     1      3        2  16  16  17  
287  ...      4        3      3     1     1      3        6  13  12  12  
329  ...      4        3      3     1     2      4        4  14  14  14  
369  ...      3        2      2     4     2      5       10  14  12  11  
10   ...      3        3      3     1     2      2        0  10   8   9  
24   ...      4        3      2     1     1      5        2  10   9   8  
38   ...      4        3      2     1     1      5        2  12  12  11  
63   ...      3        4      4     2     4      4        2  10   9   9  
105  ...      4        3      3     1     1      4       10  10  11  11  
109  ...      5        4      5     1     1      4        4  14  15  16  
122  ...      4        2      2     1     2      5        2  13  13  13  
169  ...      4        4      2     1     1      3        0  14  14  14  
274  ...      4        3      3     1     1      1        2  10  10  10  
278  ...      2        4      4     1     1      4       15   9   8   8  
303  ...      5        2      2     1     2      5        0  17  17  18  
314  ...      4        1      2     1     1      3       14  15  13  13  

[25 rows x 33 columns]